File: Test2/Options/UserOptions.vb
Imports System.IO
Imports System.Xml.Serialization

Namespace Options
	Public Class UserOptions
		Public Property PrintExportLocation As String = ""
		Public Property DxfExportLocation As String = ""

		' Feature toggles
		Public Property EnableObsoletePrint As Boolean = True

		Public Shared ReadOnly _
			OptionsFilePath As String = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
			                                         "DoyleAddinOptions.xml")

		Public Sub Save()
			Dim serializer As New XmlSerializer(GetType(UserOptions))
			Using writer As New StreamWriter(OptionsFilePath)
				serializer.Serialize(writer, Me)
			End Using
		End Sub

		Public Shared Function Load() As UserOptions
			If File.Exists(OptionsFilePath) Then
				Dim serializer As New XmlSerializer(GetType(UserOptions))
				Using reader As New StreamReader(OptionsFilePath)
					Return CType(serializer.Deserialize(reader), UserOptions)
				End Using
			Else
				Return New UserOptions()
			End If
		End Function
	End Class
End NameSpace

File: Doyle Addin C/Options/UserOptions.cs
using System;
using System.IO;
using System.Xml.Serialization;

namespace Doyle_Addin.Options
{
    public class UserOptions
    {
        public string PrintExportLocation { get; set; } = "";
        public string DxfExportLocation { get; set; } = "";

        // Feature toggles
        public bool EnableObsoletePrint { get; set; } = true;

        public static readonly string OptionsFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "DoyleAddinOptions.xml");


        public void Save()
        {
            var serializer = new XmlSerializer(typeof(UserOptions));
            using (var writer = new StreamWriter(OptionsFilePath))
            {
                serializer.Serialize(writer, this);
            }
        }

        public static UserOptions Load()
        {
            if (File.Exists(OptionsFilePath))
            {
                var serializer = new XmlSerializer(typeof(UserOptions));
                using (var reader = new StreamReader(OptionsFilePath))
                {
                    return (UserOptions)serializer.Deserialize(reader);
                }
            }
            else
            {
                return new UserOptions();
            }
        }
    }
}